File: GestureDetectorSdk/src/main/java/com/behaviosec/android/gesturedetectorsdk/AccelerometerListener.java
package com.behaviosec.android.gesturedetectorsdk;

import android.content.Context;
import android.hardware.Sensor;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.util.Log;

public class AccelerometerListener implements SensorEventListener {

    private final SensorManager sensorManager;
    private final Sensor accelerometer;

    public AccelerometerListener(Context context) {
        sensorManager = (SensorManager) context.getSystemService(Context.SENSOR_SERVICE);
        accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
    }

    public void start() {
        sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_UI);
    }

    public void stop() {
        sensorManager.unregisterListener(this);
    }

    @Override
    public void onSensorChanged(SensorEvent event) {
        float x = event.values[0];
        float y = event.values[1];
        float z = event.values[2];
        Log.d("Accelerometer", "x: " + x + ", y: " + y + ", z: " + z);
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {
        Log.d("Accelerometer", "x: ");
    }
}

File: AccelerometerTouchTrackerSdk/src/main/java/com/behaviosec/android/accelerometerTouchTrackerSdk/listeners/AccelerometerListener.kt
package com.behaviosec.android.accelerometerTouchTrackerSdk.listeners

import com.behaviosec.android.accelerometerTouchTrackerSdk.models.AccelerometerEventModel
import com.behaviosec.android.accelerometerTouchTrackerSdk.models.AccuracyChangedModel

/**
 * Accelerometer listener
 *
 * @constructor Create empty Accelerometer listener
 */
interface AccelerometerListener {
    /**
     * On sensor changed
     *
     * @param model
     */
    fun onSensorChanged(model: AccelerometerEventModel) {}

    /**
     * On accuracy changed
     *
     * @param model
     */
    fun onAccuracyChanged(model: AccuracyChangedModel) {}
}
